File: day01.c
#include <stdio.h>
#include <stdlib.h>

#define BUFFER 16384
int main(void)
{
    FILE *f = fopen("inp.txt", "r");

    if (f == NULL)
    {
        printf("Could not open file\n");
        return 1;
    }

    int first_neg_i = 1, found = 0, floor = 0, matched;
    char c;

    while ((c = fgetc(f)) != EOF)
    {
        switch (c)
        {
        case '(':
            floor++;
            break;
        case ')':
            floor--;
            break;
        }

        if (!found && floor >= 0)
        {
            first_neg_i++;
        }
        else
        {
            found = 1;
        }
    }

    printf("Floor: %d Negative floor index: %d\n", floor, first_neg_i);
}

File: day01.py
with open("inp.txt", "r") as f:
    inp = f.read().strip()

v = 0
found = False
for i, char in enumerate(inp):
    if char == '(':
        v += 1
    elif char == ')':
        v -= 1
    if not found and v < 0:
        found = True
        print(i)

print(v)